activerecord/lib/active_record/signed_id.rb

```diff
@@ -90,7 +90,7 @@ def signed_id_verifier=(verifier)
 
       # :nodoc:
       def combine_signed_id_purposes(purpose)
-        [ base_class.name.underscore, purpose.to_s ].compact_blank.join("/")
+        [ base_class.polymorphic_name.underscore, purpose.to_s ].compact_blank.join("/")
```

Comment:
Would not this change means that all valid signed id wold be considered invalid after Rails is upgraded because the `polymorphic_name` is different from `name`?

Reply:
With the [implementation](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/inheritance.rb#L200) of `polymorphic_name`, an upgrade would make previous signed ids invalid if `store_full_class_name` is set to false (default is true) and the signed id is for a namespaced model.

Do you think a method `signed_id_name` specific to this use case would be better?